### PR TITLE
Tguify traits

### LINKED ISSUE #6109: trait selection flavor text doesnt update very quickly
Submitter: tunnelflip
I dont know if its a byond thing, but moving the window seems to update it, there is a text so you can read about the trait when you hover your mouse over it but it takes to long for it to show.

### PR DESCRIPTION
**Pull request by Luxizzle**
## About the PR 

Moves traits into the main setup window for a smoother character creation flow.

Reopening of #6676 & #6136

Fixes #6109

All images are staying in dmi form and just like the spacebux purchasing menu are converted into base64.

image

## Why's this needed? 

Gets rid of bad window.

## Changelog

```changelog
(u)Luxizzle
(+)Traits have been moved into the main character setup window
```

### CODE REVIEW

**Comment by Luxizzle** [comment]
Should probably get an extra check if this doesn't alter trait limit logic, solving 2 year old merge conflicts while you are sick probably isn't the best idea.

**Comment by Luxizzle** [comment]
!merge_upstream

**Comment by Luxizzle** [comment]
!merge_upstream

**Comment by Luxizzle** [comment]
!merge_upstream

**Comment by Luxizzle** [comment]
!merge_upstream

File: code/datums/traits.dm [hunk 1]
```diff
@@ -191,13 +185,24 @@
...
 	var/requiredUnlock = null //If set to a string, the xp unlock of that name is required for this to be selectable.
 	var/cleanName = ""   //Name without any additional information.
 	var/isMoveTrait = FALSE // If TRUE, onMove will be called each movement step from the holder's mob
 	var/datum/mutantrace/mutantRace = null //If set, should be in the "species" category.
 
+	New()
+		..()
+
+		// Clean name should always be set, but if not, this replaces the (...) and [...]
+		// from the name and sets that as the clean name. Should probably just remove this and move cleanName to name as normal name never gets used anymore.
```
Comment (ZeWaka) [inline code/datums/traits.dm:199]: > Should probably just remove this and move cleanName to name as normal name never gets used anymore.

:)

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/datums/traits.dm (hunk 1 above)
Comment (Luxizzle) [reply code/datums/traits.dm:199]: This PR is already big enough to have been sitting here for a couple of weeks now.

Comment (Luxizzle) [reply code/datums/traits.dm:199]: I can open a draft PR for after this one later to replace all of these

Comment (Luxizzle) [reply code/datums/traits.dm:199]: https://github.com/goonstation/goonstation/pull/9919

**Comment by Luxizzle** [comment]
!merge_upstream

**Comment by ZeWaka** [comment]
!merge_upstream

**Pull request merged**